classes/Tools.php

```diff
@@ -461,6 +461,16 @@ public static function getCurrentUrlProtocolPrefix()
         }
     }
 
+    /**
+     * Get the current url
+     *
+     * @return string current url
+     */
+    public static function getCurrentUrl()
```

Comment:
```suggestion
    public static function getCurrentUrl(): string
```

Reply:
@jf-viguier Why not using `getHttpHost` ?

Reply:
Because getHttpHost use Configuration::get('PS_SSL_ENABLED'), Tools::usingSecureMode() is more safe: it tests real usage of ssl.

Reply:
Configuration::get('PS_SSL_ENABLED') vs Tools::usingSecureMode() is an huge issue and not the point of my pr.